scopus-import/src/main/java/no/sikt/nva/scopus/conversion/model/CorporationWithContributors.java

```diff
@@ -0,0 +1,148 @@
+import no.scopus.generated.OrganizationTp;
+import no.unit.nva.expansion.model.cristin.CristinOrganization;
+import no.unit.nva.language.LanguageMapper;
+import no.unit.nva.model.Corporation;
+import no.unit.nva.model.Organization;
+import no.unit.nva.model.UnconfirmedOrganization;
+import org.apache.tika.langdetect.optimaize.OptimaizeLangDetector;
+
+public class CorporationWithContributors {
+
+    public static final String NORWAY = "norway";
+
+    private AuthorGroupTp scopusAuthors;
+
+    private List<CristinOrganization> cristinOrganizations;
+    private URI cristinOrganizationId;
+
+    public static String getLanguageIso6391Code(String textToBeGuessedLanguageCodeFrom) {
+        var detector = new OptimaizeLangDetector().loadModels();
+        var result = detector.detect(textToBeGuessedLanguageCodeFrom);
+        return result.isReasonablyCertain() ? getIso6391LanguageCodeForSupportedNvaLanguage(result.getLanguage())
+                   : ENGLISH.getIso6391Code();
+    }
+
+    public AuthorGroupTp getScopusAuthors() {
+        return scopusAuthors;
+    }
+
+    public void setScopusAuthors(AuthorGroupTp scopusAuthors) {
+        this.scopusAuthors = scopusAuthors;
+    }
+
+    public List<CristinOrganization> getCristinOrganizations() {
+        return isNull(cristinOrganizations)
+                   ? List.of()
+                   : cristinOrganizations;
+    }
+
+    public void setCristinOrganizations(List<CristinOrganization> cristinOrganizations) {
+        this.cristinOrganizations = cristinOrganizations;
+    }
+
+    public URI getCristinOrganizationId() {
+        return cristinOrganizationId;
+    }
+
+    public void setCristinOrganizationId(URI cristinId) {
+        this.cristinOrganizationId = cristinId;
+    }
+
+    public List<Corporation> toCorporations() {
+        return isNull(cristinOrganizations) || cristinOrganizations.isEmpty()
+                   ? generateCorporationFromAuthorGroupTp()
+                   : generateCorporationFromCristinOrganization();
+    }
+
+    public Builder copy() {
+        return new Builder().withAuthorGroupTp(scopusAuthors)
+                   .withCristinOrganizationId(cristinOrganizationId)
+                   .withCristinCorporations(cristinOrganizations);
+    }
+
+    private static boolean isNotNorway(Map<String, String> labels) {
+        return labels.values().stream().map(String::toLowerCase).noneMatch(NORWAY::equals);
+    }
+
+    private static String toOrganizationName(List<OrganizationTp> organizationTps) {
+        return organizationTps.stream()
+                   .map(organizationTp -> extractContentString(organizationTp.getContent()))
+                   .collect(Collectors.joining(AFFILIATION_DELIMITER));
+    }
+
+    private static String getIso6391LanguageCodeForSupportedNvaLanguage(String possiblyUnsupportedLanguageIso6391code) {
+        var language = LanguageMapper.getLanguageByIso6391Code(possiblyUnsupportedLanguageIso6391code);
```

Comment:
IIRC, it doesn't return null, it returns `und`

Reply:
Actually the build fails here are null pointer exception when I changed.